src/PaginationCursor.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSauce\EventSourcing;
+
+interface PaginationCursor
+{
+    public function toString(): string;
+
+    public static function fromString(string $cursor): static;
+
+    public function isAtStart(): bool;
```

Comment:
I don't feel this is resolved. What if we had:

```php
public function start(): int|string;

public function limit(): int;
```

As part of this interface? That would allow for the cursor to be offset/limit based, or use a different kind of start control (date based?) while still covering the majority of us that are going to use offset/limit.

Reply:
I think we can leave the interface to contain only what is used on the outside of the implementation and let the specific cursors define their own specific getters. The surrounding code doesn't seem to need to know the start, seems more important for the specific repository, so adding it to the interface doesn't provide much value right now.